test/mod/test_monitor.py

```diff
@@ -66,28 +63,6 @@ def log(self, message: str):
 
 
 class TestMonitor(unittest.TestCase):
-    def test_log_monitor_api(self):
-        # Basic log and API integration check
-        with tempfile.TemporaryDirectory() as tmpdir:
-            args = {"foo": "bar"}
-            path = os.path.join(tmpdir, "osbuild-api")
-            logfile = os.path.join(tmpdir, "log.txt")
-
-            with open(logfile, "w") as log:
-                api = API(args, LogMonitor(log.fileno()), socket_address=path)
-                with api as api:
-                    p = mp.Process(target=echo, args=(path, ))
-                    p.start()
-                    p.join()
-                    self.assertEqual(p.exitcode, 0)
-                output = api.output  # pylint: disable=no-member
-                assert output
-
-            self.assertEqual(json.dumps(args), output)
-            with open(logfile) as f:
-                log = f.read()
-            self.assertEqual(log, output)
-
```

Comment:
If we drop this entire test, I think you can drop the `echo` function as well. It is only used by this test, right?